specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-02-01/managedClusters.json

```diff
@@ -1116,6 +1160,40 @@
       },
       "description": "AgentPoolType represents types of an agent pool"
     },
+    "AgentPoolListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AgentPool"
+          },
+          "description": "The list of agent pools."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of agent pool results.",
+          "readOnly": true
+        }
+      },
+      "description": "The response from the List Agent Pools operation."
+    },
+    "AgentPool": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
```

Comment:
As far as I can tell, AgentPool is not a legitimate child resource as far as ARM is concerned, it is just a property within the resource. Is that correct?

Reply:
That is correct. Agentpool is part of the managedCluster resourece right now. But we are separating it out to be an untracked child resource of managedCluster. 
Note: I did not include the CRUD APIs for agentpool yet because the API is not deployed. This PR is just to include the data model changes in the new API version.